app/scripts/metamask-controller.js

```diff
@@ -4804,6 +4811,20 @@ export default class MetamaskController extends EventEmitter {
     }
   }
 
+  async socialSyncKeyringEncryptionKey() {
+    const isSocialLoginFlow = this.onboardingController.getIsSocialLoginFlow();
+    if (!isSocialLoginFlow) {
+      return;
+    }
```

Comment:
do you actually need to check for this here? in most places you already check for it in the caller. it's also a bit weird that this function does nothing when it is called in the wrong place. i think it's better to call it only intentionally.